commons-test/src/main/java/org/infinispan/commons/test/EnvironmentCheck.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.commons.test;
+
+/**
+ * Verifies that the tests are being run with a JVM version which matches the supplied regex
+ *
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 10.0
+ **/
+public class EnvironmentCheck {
+
+   public static final void checkJVMVersion() {
+      String jvmVersionRegex = System.getProperty("test.jvm.version.regex");
+      if (jvmVersionRegex != null) {
+         if (!System.getProperty("java.version").matches(jvmVersionRegex)) {
+            System.err.printf("JVM version '%s' does not match '%s'\n", System.getProperty("java.version"), jvmVersionRegex);
+            System.exit(1);
```

Comment:
How is this reported in Jenkins? I still haven't found a good alternative for reporting thread leaks...

Reply:
I have changed it to throw a RuntimeException. It fails the build locally.